pkg/yttlibrary/overlay/array_item_match_annotation.go

```diff
@@ -66,7 +66,12 @@ func (a ArrayItemMatchAnnotation) Indexes(leftArray *yamlmeta.Array) ([]int, err
 		return nil, err
 	}
 
-	return idxs, a.expects.Check(len(idxs))
+	nodes := MatchingNodes{}
```

Comment:
@cppforlife one "draft" aspect of the PR is that — all things being equal — would expect `MatchNodes` to return the collection of matching nodes.  However, there is downstream logic that expects a collection of indexes.

A question I'm exploring: what made passing around indexes of matches a suitable tactic, to date? ... does this PR/feature now make passing around references _more_ suitable (or not)?

Reply:
indexes are used to reference to a node within its parent (eg array item within array). since things can be removed you need stable ref to what to remove.